### PR TITLE
[jvm-packages] Making the Java library loader emit helpful error messages on missing dependencies

### PR DESCRIPTION
**Pull request by Craigacp**
As discussed in #6767 and #6793 I've updated the error messages thrown when the native xgboost library fails to load in Java. It now exposes the OS and architecture and emits a different error message depending on the platform, requesting that the user installs libomp in various forms. I also added a note saying the Linux one requires glibc, as we've hit this issue when deploying XGBoost into small containers based on Alpine Linux which uses musl libc.

An example of the new error message on macOS under Java 11 is below:

```
[04/30/21 10:17:37:406] SEVERE NativeLibLoader.initXGBoost Failed to load xgboost4j due to missing native dependencies for platform macos/x86_64, this is likely due to a missing OpenMP dependency
[04/30/21 10:17:37:406] SEVERE NativeLibLoader.initXGBoost You may need to install 'libomp.dylib', via `brew install libomp` or similar
Exception in thread "main" java.lang.UnsatisfiedLinkError: /private/var/folders/w8/tzq8tdzj42d5qjhwynpdph8c00h820/T/libxgboost4j14305990415700311532.dylib: dlopen(/private/var/folders/w8/tzq8tdzj42d5qjhwynpdph8c00h820/T/libxgboost4j14305990415700311532.dylib, 1): Library not loaded: /usr/local/opt/libomp/lib/libomp.dylib
  Referenced from: /private/var/folders/w8/tzq8tdzj42d5qjhwynpdph8c00h820/T/libxgboost4j14305990415700311532.dylib
  Reason: image not found
	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2430)
	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2487)
	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2684)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2617)
	at java.base/java.lang.Runtime.load0(Runtime.java:765)
	at java.base/java.lang.System.load(System.java:1835)
	at ml.dmlc.xgboost4j.java.NativeLibLoader.loadLibraryFromJar(NativeLibLoader.java:177)
	at ml.dmlc.xgboost4j.java.NativeLibLoader.initXGBoost(NativeLibLoader.java:130)
	at ml.dmlc.xgboost4j.java.XGBoostJNI.<clinit>(XGBoostJNI.java:34)
	at ml.dmlc.xgboost4j.java.DMatrix.<init>(DMatrix.java:109)
```

### CODE REVIEW

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java [hunk 1]
```diff
@@ -34,65 +34,127 @@
...
+     */
+    static OS detectOS() {
+      String os = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
+      if (os.contains("mac") || os.contains("darwin")) {
+        return MACOS;
+      } else if (os.contains("win")) {
+        return WINDOWS;
+      } else if (os.contains("nux")) {
+        return LINUX;
+      } else if (os.contains("sunos")) {
```
Comment (trivialfis) [inline jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java:65]: Not sure if the BSD family should be added here.  They do package the Python library.

Comment (Craigacp) [reply jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java:65]: It should be straightforward to add. Are there any other platforms? I didn't bother with things like AIX or POWER, but the JVM will run on them.

**Review by hcho3 — APPROVED** [review]

File: jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java (hunk 1 above)
Comment (trivialfis) [reply jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java:65]: Nah, let's do it when the case arises.  We don't test it on *BSD anyway.

**Review by trivialfis — APPROVED** [review]
Thanks for the PR!

**Pull request merged**